src/app/public/modules/ag-grid/ag-grid-data-manager-adapter.directive.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  AfterContentInit,
+  ChangeDetectorRef,
+  ContentChildren,
+  Directive,
+  Input,
+  OnDestroy,
+  QueryList
+} from '@angular/core';
+
+import {
+  Subject,
+  Subscription
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  AgGridAngular
+} from 'ag-grid-angular';
+
+import {
+  ColumnMovedEvent,
+  RowSelectedEvent
+} from 'ag-grid-community';
+
+import {
+  SkyDataManagerService,
+  SkyDataManagerSortOption,
+  SkyDataManagerState,
+  SkyDataViewConfig
+} from '@skyux/data-manager';
+
+@Directive({
+  selector: '[skyAgGridDataManagerAdapter]'
+})
+export class SkyAgGridDataManagerAdapterDirective implements AfterContentInit, OnDestroy {
+
+  @Input()
+  private viewId: string;
+
+  @ContentChildren(AgGridAngular, { descendants: true })
+  public agGridList: QueryList<AgGridAngular>;
+
+  private currentAgGrid: AgGridAngular;
+  private dataStateSub: Subscription;
+  private viewConfig: SkyDataViewConfig;
+  private ngUnsubscribe = new Subject();
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private dataManagerSvc: SkyDataManagerService) { }
+
+  public ngAfterContentInit(): void {
+    this.viewConfig = this.dataManagerSvc.getViewById(this.viewId);
+    this.checkForAgGrid();
+
+    this.agGridList.changes
+    .pipe(takeUntil(this.ngUnsubscribe))
+    .subscribe(() => this.checkForAgGrid());
+  }
+
+  public ngOnDestroy() {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  private checkForAgGrid(): void {
+    const agGridCount = this.agGridList.length;
+
+    /* istanbul ignore else */
+    if (agGridCount > 1) {
+      this.registerAgGrid();
+      console.warn(
+        'More than one ag-grid child component was found. Using the first ag-Grid.'
+      );
+    } else if (agGridCount === 0) {
+      this.unregisterAgGrid();
+    } else if (this.agGridList.first !== this.currentAgGrid) {
+      this.registerAgGrid();
+    }
+  }
+
+  private unregisterAgGrid(): void {
+    this.currentAgGrid = undefined;
+
+    /* istanbul ignore if */
+    if (this.dataStateSub) {
+      this.dataStateSub.unsubscribe();
+    }
+  }
+
+  private registerAgGrid(): void {
+    this.unregisterAgGrid();
+
+    const agGrid = this.agGridList.first;
+
+    this.currentAgGrid = agGrid;
+
+    agGrid.gridReady
+    .pipe(takeUntil(this.ngUnsubscribe))
```

Comment:
It looks like this line didn't get indented.

Reply:
woops